tests/org.jboss.tools.docker.ui.bot.test/src/org/jboss/tools/docker/ui/bot/test/AbstractDockerBotTest.java

```diff
@@ -112,59 +105,109 @@ protected static void openDockerPerspective(){
+	protected static void deleteConnection() {
+		new DockerExplorer().deleteConnection(getDockerServer());
 	}
 	
 	protected static void deleteImage(String imageName) {
-		deleteImage(System.getProperty("dockerServerURI"),imageName);
+		new DockerExplorer().deleteImage(getDockerServer(),imageName);
 	}
 	
-	protected static void deleteImage(String dockerServer, String imageName) {
-		DockerExplorer de = new DockerExplorer();
-		de.open();
-		ConnectionItem dockerConnection = de.getConnection(dockerServer);
-		TreeItem dc = dockerConnection.getTreeItem();
-		dc.select();
-		dc.expand();
-		TreeItem imagesItem = dc.getItem("Images");
-		imagesItem.select();
-		imagesItem.expand();
-		for (TreeItem item : imagesItem.getItems()) {
-			if (item.getText().contains(imageName)) {
-				item.select();
-				new ContextMenu("Remove").select();
-				new WaitUntil(new ShellWithTextIsActive("Confirm Remove Image"), TimePeriod.NORMAL);
-				new PushButton("OK").click();
-				new WaitWhile(new JobIsRunning(), TimePeriod.NORMAL);
+	protected static boolean imageIsDeployed(String imageName){
+		return new DockerExplorer().imageIsDeployed(getDockerServer(),imageName);
+	}
+	
+	protected static void deleteContainer(String containerName) {
+		new DockerExplorer().deleteContainer(getDockerServer(),containerName);
+	}
+	
+	protected void pullImage(String imageName){
+		new DockerExplorer().pullImage(getDockerServer(),imageName);
+	}
+	
+	protected void pullImage(String registry,String imageName){
+		new DockerExplorer().pullImage(getDockerServer(),registry,imageName);
+	}
+	
+	
+	protected void checkBrowserForErrorPage(BrowserView browser) {
+		checkBrowserForErrorPage(browser,null);
+	}
+	
+	protected void checkBrowserForErrorPage(BrowserView browser, String url) {
+		ConsoleView consoleView = new ConsoleView();
+		consoleView.open();
+		if(browser.getText().contains("Unable")||browser.getText().contains("404")){
+			if(url==null){
+				browser.refreshPage();
+			}else{
+				browser.openPageURL(url);
 			}
 		}
+		new WaitWhile(new JobIsRunning());
+		assertFalse("Browser contains text 'Status 404'\n Console output:\n" + consoleView.getConsoleText(),
+				browser.getText().contains("Status 404") || browser.getText().contains("404 - Not Found")|| browser.getText().contains("Unable")|| browser.getText().contains("Forbidden")|| browser.getText().contains("404"));
+		assertFalse(
+				"Browser contains text 'Error processing request'\n Console output:\n" + consoleView.getConsoleText(),
+				browser.getText().contains("Error processing request"));
 	}
 	
-	protected static boolean imageIsDeployed(String imageName){
-		return imageIsDeployed(System.getProperty("dockerServerURI"),imageName);
+	protected String createURL(String tail){
+		String dockerServerURI = System.getProperty("dockerServerURI");
+		String serverURI = dockerServerURI!=null&&!dockerServerURI.isEmpty()?System.getProperty("dockerServerURI").replaceAll("tcp://","http://"):"http://localhost:1234";
```

Comment:
I know, that this line is matter of taste. I personally prefer "if" to ? : notation (i think "if" is more readable). But it's up to you, if you want to leave this line as it is. If that's the case, please use auto formatter ;-)